integrations/astra/tests/test_document_store.py

```diff
@@ -172,6 +172,40 @@ def test_delete_documents_more_than_twenty_delete_ids(self, document_store: Astr
         # No Document has been deleted
         assert document_store.count_documents() == 0
 
+    def test_filter_documents_nested_filters(self, document_store, filterable_docs):
+        """
+        Test filter_documents() with nested filters. Two documents from the filterable_docs mixin
+        should be found. If those documents change, this test has to change as well.
+        """
+        filter_criteria = {
+            "operator": "AND",
+            "conditions": [
+                {"field": "meta.page", "operator": ">=", "value": "100"},
+                {
+                    "operator": "OR",
+                    "conditions": [
+                        {"field": "meta.chapter", "operator": "==", "value": "abstract"},
+                        {"field": "meta.chapter", "operator": "==", "value": "intro"},
+                    ],
+                },
+            ],
+        }
+
+        document_store.write_documents(filterable_docs)
+        result = document_store.filter_documents(filters=filter_criteria)
+
+        expected_docs = [
+            d
+            for d in filterable_docs
+            if d.meta.get("page") >= "100" and (d.meta.get("chapter") == "abstract" or d.meta.get("chapter") == "intro")
+        ]
+        assert len(expected_docs) == 2
```

Comment:
`assert_documents_are_equal ` is comparing the results. So this is not needed.
```suggestion
```

Reply:
ah i see! i thought it might be a more useful test to ensure that we actually get documents from the filters instead of just comparing the results since it's possible that the filters can get no documents and the comparison would still pass. what do you think?

Reply:
That wouldn't be necessary as we are comparing two methods of retrieving the documents (using astra filters, and through simple operators). Passing these to `assert_documents_are_equal `  verifies both the number and content of the retrieved documents are equal.